src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
+    const payload = JSON.stringify(request.body);
+
+    if (!signature) {
```

Comment:
PullProofAI

Consider adding additional validation checks for the `signature` variable before proceeding with further processing to ensure data integrity and prevent potential vulnerabilities.
```diff
+ if (typeof signature !== 'string' || !signature.trim()) {
+     throw new UnauthorizedException('Invalid signature format');
+ }
```


<!-- This is an auto-generated comment by PullProofAI -->